src/keyring/ed25519/signing_key.rs

```diff
@@ -6,70 +7,120 @@ use signature::Signer;
 type SigningKeyBytes = [u8; SigningKey::BYTE_SIZE];
 
 /// Ed25519 signing key.
-#[derive(Clone, Debug)]
-pub struct SigningKey(ed25519_consensus::SigningKey);
+pub enum SigningKey {
+    /// Ed25519 signing key.
+    Ed25519(ed25519_consensus::SigningKey),
+    /// Ed25519 expanded signing key.
+    Ed25519Expanded(ExpandedSecretKey),
+}
```

Comment:
My preference would still be to unify these somehow. You should be able to initialize an `ExpandedSecretKey` from either a seed or its SHA-512 expansion.

Then this could just be:

```suggestion
pub struct SigningKey(ExpandedSecretKey);
```